File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        maxl = 1
        l = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                l += 1
            else:
                l = 1
            maxl = max(maxl, l)
        minl = 1
        l = 1
        for i in range(1, len(nums)):
            if nums[i] < nums[i-1]:
                l += 1
            else:
                l = 1
            minl = max(minl, l)

        # print(maxl, minl)
        return max(maxl, minl)

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int minL(1), maxL(1),ans(1);
        for(int i=1; i<nums.size(); i++){
            if(nums[i]>nums[i-1]){
                maxL++;
                minL =1;
            }
            else if(nums[i]<nums[i-1]){
                minL++;
                maxL =1;
            }
            else{
                minL=1;maxL=1;
            }
            ans = max({ans,maxL,minL});
        }
        return ans;
    }
};